devicemodel/hw/pci/virtio/virtio_net.c

```diff
@@ -940,8 +940,11 @@ virtio_net_init(struct vmctx *ctx, struct pci_vdev *dev, char *opts)
 		return -1;
 	}
 
-	vtopts = tmp = strdup(opts);
-	if (strncmp(tmp, "tap", 3) == 0) {
+	if (opts != NULL) {
+		vtopts = tmp = strdup(opts);
+	}
+
+	if ((tmp != NULL) && (strncmp(tmp, "tap", 3) == 0)) {
```

Comment:
The tmp is impossible to be NULL as opts has checked already.

Reply:
Sometimes the strdup will return NULL if malloc failed.